pkg/util/version/stream_test.go

```diff
@@ -42,46 +42,59 @@ func TestGetUpgradeStream(t *testing.T) {
 	}
 
 	for _, tt := range []struct {
-		name string
-		v    *Version
-		want Stream
-		err  error
+		name     string
+		v        *Version
+		want     Stream
+		upgradeY bool
+		wantErr  string
 	}{
 		{
-			name: "upgrade when x.Y is lower than expected",
+			name: "upgrade z when x.Y is lower than expected",
 			v:    NewVersion(4, 3, 17),
 			want: stream43,
 		},
 		{
-			name: "no upgrade when x.Y is higher than exected",
-			v:    NewVersion(4, 3, 19),
-			err:  fmt.Errorf("not upgrading: cvo desired version is 4.3.19"),
+			name:    "don't upgrade z when x.Y is higher than expected",
+			v:       NewVersion(4, 3, 19),
+			wantErr: "not upgrading: CVO desired version is 4.3.19",
 		},
 		{
-			name: " when X.y id lower than exected",
+			name: "upgrade z when X.y is lower than expected",
 			v:    NewVersion(4, 4, 2),
 			want: stream44,
 		},
 		{
-			name: "no upgrade when X.y is higher than expected",
-			v:    NewVersion(4, 4, 9),
-			err:  fmt.Errorf("not upgrading: cvo desired version is 4.4.9"),
+			name:    "don't upgrade z when X.y is higher than expected",
+			v:       NewVersion(4, 4, 9),
+			wantErr: "not upgrading: CVO desired version is 4.4.9",
 		},
 		{
-			name: "cvo error",
-			v:    NewVersion(4, 5, 1),
-			err:  fmt.Errorf("not upgrading: stream not found 4.5.1"),
+			name:     "upgrade y when allowed and X.y is equal to current y version",
+			v:        NewVersion(4, 3, 18),
+			upgradeY: true,
+			want:     stream44,
 		},
 		{
-			name: "error",
-			v:    NewVersion(5, 5, 1),
-			err:  fmt.Errorf("not upgrading: stream not found 5.5.1"),
+			name:    "don't upgrade y when not allowed",
+			v:       NewVersion(4, 3, 18),
+			wantErr: fmt.Sprintf("not upgrading: Y version upgrade to %s was not allowed", stream44.Version.String()),
+		},
+		{
+			name:    "CVO error",
+			v:       NewVersion(4, 5, 1),
+			wantErr: "not upgrading: stream not found 4.5.1",
+		},
+		{
+			name:    "error",
+			v:       NewVersion(5, 5, 1),
+			wantErr: "not upgrading: stream not found 5.5.1",
 		},
 	} {
 		t.Run(tt.name, func(t *testing.T) {
 			Streams = []Stream{stream43, stream44}
```

Comment:
I think we need to extend this test to third version:
If Y is allowed, I want to be upgraded to Y+1 and make sure I will not be upgraded to Y+2 if it is available. 
I think logic is right, but tests just to be sure. For this we might need to override 
So we have `Streams = []Stream{stream43, stream44, stream45}` 

As once 4.5 lands we will have more than 2 streams in the list.

Reply:
@mjudeikis Added